torchvision/prototype/transforms/_geometry.py

```diff
@@ -866,3 +866,20 @@ def forward(self, *inputs: Any) -> Any:
             )
 
         return super().forward(*inputs)
+
+
+class RandomResize(Transform):
+    def __init__(self, min_size: int, max_size: Optional[int] = None):
+        super().__init__()
+        self.min_size = min_size
+        self.max_size = max_size if max_size is not None else min_size
```

Comment:
Triggering this default actually makes this transform a no-op. In the reference scripts `max_size` is always passed:

https://github.com/pytorch/vision/blob/cac4e228c9ca9e7564cb34406e7ebccfdd736976/references/segmentation/presets.py#L10

https://github.com/pytorch/vision/blob/cac4e228c9ca9e7564cb34406e7ebccfdd736976/references/segmentation/presets.py#L31

(why are we using a random transformation in the evaluation transforms? Doesn't that mean we get different metrics every time we run this?)

We should probably get rid of this behavior, since it makes little sense to use this transform in the current state without passing a `max_size`.

Reply:
I would say it's not no-op but rather falls back to Resize. If I had to guess, the original author didn't want to provide another Resize wrapper to handle Images and Masks at the same time, so they hacked this one. I agree we can use the standard `Resize()` of the new API during inference. So I'm also in favour of getting rid of this behaviour.